<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <title>CRAN Task View: Causal Inference</title>
  <link rel="stylesheet" type="text/css" href="../CRAN_web.css" />
  <meta http-equiv="content-type" content="text/html; charset=UTF-8" />
  <meta name="citation_title" content="CRAN Task View: Causal Inference" />
  <meta name="citation_author" content="Imke Mayer, Pan Zhao and Julie Josse" />
  <meta name="citation_publication_date" content="2021-08-18" />
  <meta name="citation_public_url" content="https://misscausal.gitlabpages.inria.fr/misscausal.gitlab.io/software.html" />
  <meta name="DC.title" content="CRAN Task View: Causal Inference" />
  <meta name="DC.creator" content="Imke Mayer, Pan Zhao and Julie Josse" />
  <meta name="DC.issued" content="2021-08-18" />
  <meta name="DC.identifier" content="https://misscausal.gitlabpages.inria.fr/misscausal.gitlab.io/software.html" />
</head>

<body>
  <h2>CRAN Task View: Causal Inference</h2>
  <table summary="CausalInference task view information">
    <tr><td valign="top"><b>Maintainer:</b></td><td>Imke Mayer, Pan Zhao and Julie Josse</td></tr>
    <tr><td valign="top"><b>Contact:</b></td><td>&#x6d;&#x61;&#x79;&#x65;&#x72;&#x2e;&#x69;&#x6d;&#x6b;&#x65;&#x20;&#x61;&#x74;&#x20;&#x67;&#x6d;&#x61;&#x69;&#x6c;&#x2e;&#x63;&#x6f;&#x6d;</td></tr>
    <tr><td valign="top"><b>Version:</b></td><td>2021-08-18</td></tr>
    <tr><td valign="top"><b>URL:</b></td><td><a href="https://misscausal.gitlabpages.inria.fr/misscausal.gitlab.io/software.html">https://misscausal.gitlabpages.inria.fr/misscausal.gitlab.io/software.html</a></td></tr>
  </table>

    <div>
      <p>
        Causal inference can be seen as a subfield of statistical analysis that addresses questions about effects of a treatment, intervention or policy on some target over a given sample or population. Under certain model assumptions, causal inferences can be carried out by fitting simple regression models or combining several regression models in a specific way as will be sketched out later. For observational data, additional untestable assumptions have to be made to (non-parametrically) identify causal effects.
      </p>
      <p>
        There do not exist basic R functions to do causal analyses straight away, but many methods - more or less complex - are implemented in different packages on CRAN, which we structure into main topics:
      </p>
      <ul>
        <li>
          <a href="#rct">
            Methods for experimental data
          </a>
        </li>
        <li>
          <a href="#ate">
            Point estimation of treatment effects
          </a>
        </li>
        <li>
          <a href="#hte">
            Heterogeneous treatment effect estimation
          </a>
        </li>
        <li>
          <a href="#policy">
            Policy learning and dynamic treatment regimes
          </a>
        </li>
        <li>
          <a href="#dag">
            Do-calculus and causal discovery
          </a>
        </li>
        <li>
          <a href="#data">
            Specific types of data
          </a>
        </li>
        <li>
          <a href="#applications">
            Specific application fields
          </a>
        </li>
      </ul>
      <p>
        If you think that we missed some important packages in this list, please contact the maintainers.
      </p>
      <p>
        <a name="rct">
          <strong>
            Methods for randomized controlled trial (RCT) and other experimental data
          </strong>
        </a>
      </p>
      <ul>
        <li>
          <i>
            Construction of experimental designs
          </i>
          is implemented in
<a href="../packages/blocksdesign/index.html">blocksdesign</a>
          (blocks for general factorial treatment designs),
<a href="../packages/BCHM/index.html">BCHM</a>
          (Bayesian cluster hierarchical model design for multiple subgroup basket trials),
<a href="../packages/Boptbd/index.html">Boptbd</a>
          (Bayesian optimal block designs under linear mixed effects model),
<a href="../packages/seqDesign/index.html">seqDesign</a>
          (sequential design of randomized two-stage treatment efficacy trials with time-to-event endpoints). Many other tools and packages exist for designing experiments and clinical trials, we refer to the
<a href="ExperimentalDesign.html">ExperimentalDesign</a>
          and
<a href="ClinicalTrials.html">ClinicalTrials</a>
          CRAN Task Views.
        </li>
        <li>
          <i>
            Tests based on pairwise comparisons
          </i>
          are provided in
<a href="../packages/BuyseTest/index.html">BuyseTest</a>.
        </li>
        <li>
          <i>
            Regression models
          </i>
          where the causal estimand is defined as a coefficient of a regression model are implemented in the packages
<a href="../packages/allestimates/index.html">allestimates</a>.
        </li>
        <li>
          <i>
            Evaluation of RCTs
          </i>
          is provided in
<a href="../packages/experiment/index.html">experiment</a>
          (various statistical methods),
<a href="../packages/eefAnalytics/index.html">eefAnalytics</a>
          (Frequentist and Bayesian multilevel models),
<a href="../packages/ipcwswitch/index.html">ipcwswitch</a>
          (IPW adapted to treatment switch in an RCT).
        </li>
        <li>
          <i>
            Posterior analysis tools
          </i>
          are implemented in
<a href="../packages/beanz/index.html">beanz</a>
          (Bayesian HTE models),
<a href="../packages/cjoint/index.html">cjoint</a>
          (conjoint analysis for survey experiments).
        </li>
        <li>
          Inference in randomized controlled trials with death and missingness is implemented in
<a href="../packages/idem/index.html">idem</a>.
        </li>
        <li>
          Inverse probability of censoring weights to handle treatment switches when estimating causal treatment effect in randomized clinical trials is implemented in
<a href="../packages/ipcwswitch/index.html">ipcwswitch</a>
        </li>
        <li>
          Design and analysis of 2-stage preference trials is implemented in
<a href="../packages/preference/index.html">preference</a>.
        </li>
        <li>
          For two-stage randomized experiments with interference and binary outcome
<a href="../packages/interferenceCI/index.html">interferenceCI</a>
          implements treatment effect estimation methods.
        </li>
        <li>
          <i>
            In case of non-compliance
          </i>,
<a href="../packages/rpsftm/index.html">rpsftm</a>
          uses g-estimation to estimate the causal effect of a treatment in a two-armed randomised control trial where non-compliance exists and is measured, under an assumption of an accelerated failure time model and no unmeasured confounders.
        </li>
        <li>
          A time series causal inference model for RCT under spillover effect is implemented in
<a href="../packages/SPORTSCausal/index.html">SPORTSCausal</a>.
        </li>
        <li>
          Design and analysis of clinical non-inferiority or superiority trials with active and placebo control is implemented in
<a href="../packages/ThreeArmedTrials/index.html">ThreeArmedTrials</a>.
        </li>
      </ul>
      <p>
        <a name="ate">
          <strong>
            Point estimation of treatment effects
          </strong>
        </a>
      </p>
      <ul>
        <li>
          <i>
            Regression models
          </i>
          where the causal estimand is a regression parameter are implemented in
<a href="../packages/estimatr/index.html">estimatr</a>,
<a href="../packages/CausalGAM/index.html">CausalGAM</a>
          (using generalized additive models),
<a href="../packages/sampleSelection/index.html">sampleSelection</a>
          (two-step and maximum likelihood estimation of Heckman-type sample selection models),
<a href="../packages/BCEE/index.html">BCEE</a>
          (Bayesian causal effect estimation for binary or continuous exposures and outcomes),
<a href="../packages/borrowr/index.html">borrowr</a>
          (Bayesian PATE estimation for multiple exchangeable data sources),
<a href="../packages/causaldrf/index.html">causaldrf</a>
          (average causal dose response functions),
<a href="../packages/hdm/index.html">hdm</a>
          (efficient estimators with uniformly valid confidence intervals, it assumes approximately sparse models for high-dimensional settings).
        </li>
        <li>
          <i>
            Matching
          </i>
          methods are implemented in
<a href="../packages/Matching/index.html">Matching</a>
          for multivariate and propensity score matching and for finding optimal balance based on a genetic search algorithm,
<a href="../packages/MatchIt/index.html">MatchIt</a>
          for selecting matched samples of the original treated and control groups with similar covariate distributions,
<a href="../packages/MatchThem/index.html">MatchThem</a>
          for the pre-processing techniques of matching and weighting multiply imputed datasets,
<a href="../packages/optmatch/index.html">optmatch</a>
          for distance based bipartite matching using the RELAX-IV minimum cost flow solver,
<a href="../packages/CMatching/index.html">CMatching</a>
          (for cluster data),
<a href="../packages/FLAME/index.html">FLAME</a>
          (almost-matching-exactly via learned weighted Hamming distance).
        </li>
        <li>
          <i>
            Inverse propensity weighting (IPW)
          </i>
          (also known as inverse probability of treatment weighting, IPTW) is implemented in
<a href="../packages/ipw/index.html">ipw</a>,
<a href="../packages/causalweight/index.html">causalweight</a>,
<a href="../packages/estimatr/index.html">estimatr</a>,
<a href="../packages/clusteredinterference/index.html">clusteredinterference</a>
          and
<a href="../packages/inferference/index.html">inferference</a>
          (the latter two both making interference assumptions) and
<a href="../packages/hrIPW/index.html">hrIPW</a>
          (under a Cox proportional hazards model).
<a href="../packages/ipwCoxCSV/index.html">ipwCoxCSV</a>
          implements a corrected sandwich variance estimation for the IPW Cox model.
<a href="../packages/ipwErrorY/index.html">ipwErrorY</a>
          implements correction methods for the IPW estimation with misclassified binary outcomes.
<a href="../packages/autoCovariateSelection/index.html">autoCovariateSelection</a>
          offers automated covariate selection for high-dimensional propensity scores. In case of
          <i>
            ordinal or multinomial treatment
          </i>,
<a href="../packages/GPSCDF/index.html">GPSCDF</a>
          allows to estimate generalized propensity score cumulative distribution functions.
        </li>
        <li>
          <i>
            Other balancing score weighting methods
          </i>
          for balancing covariate distributions are available in
<a href="../packages/CBPS/index.html">CBPS</a>
          (for binary and multivalued exposures, optionally in longitudinal settings).
<a href="../packages/PSweight/index.html">PSweight</a>
          implements a set of tools for propensity score weighting analysis of observational studies and randomized trials.
<a href="../packages/pstest/index.html">pstest</a>
          provides specification tests for parametric propensity score models.
<a href="../packages/twang/index.html">twang</a>
          provides a set of functions for propensity score estimating and weighting, nonresponse weighting, and diagnosis of the weights (of nonequivalent groups).
<a href="../packages/twangContinuous/index.html">twangContinuous</a>
          provides functions for propensity score estimation and weighting for continuous exposures.
        </li>
        <li>
          <i>
            G-formula and other conditional outcome regression based methods
          </i>
          are supported in the packages
<a href="../packages/gfoRmula/index.html">gfoRmula</a>
          (also for time-varying treatment and confounding),
<a href="../packages/EffectLiteR/index.html">EffectLiteR</a>
          (based on structural equation modeling), and
<a href="../packages/endoSwitch/index.html">endoSwitch</a>
          (maximum likelihood estimation of endogenous switching regression models).
        </li>
        <li>
          <i>
            Doubly robust methods
          </i>
          are implemented in
<a href="../packages/grf/index.html">grf</a>
          (function
<tt>average_treatment_effect</tt>) and in
<a href="../packages/causalweight/index.html">causalweight</a>.
        </li>
        <li>
          <i>
            Targeted learning
          </i>
          methods are available in
<a href="../packages/drtmle/index.html">drtmle</a>,
<a href="../packages/tmle/index.html">tmle</a>, and
<a href="../packages/ltmle/index.html">ltmle</a>
          (for longitudinal data).
        </li>
        <li>
          <i>
            Difference in differences methods
          </i>
          are implemented in
<a href="../packages/DRDID/index.html">DRDID</a>
          (doubly robust estimators with two choices for nuisance function estimation),
<a href="../packages/bacondecomp/index.html">bacondecomp</a>
          (using the Goodman-Bacon decomposition to allow for variation in treatment timing),
<a href="../packages/did/index.html">did</a>
          (for cases with more than two periods and with variation in treatment timing), and in
<a href="../packages/qte/index.html">qte</a>.
        </li>
        <li>
          <i>
            Quantile treatment effects
          </i>
          can be estimated using the
<a href="../packages/qte/index.html">qte</a>,
<a href="../packages/Counterfactual/index.html">Counterfactual</a>
          and
<a href="../packages/grf/index.html">grf</a>
          packages.
        </li>
        <li>
          <i>
            Odds ratio
          </i>
          estimation and power calculation for the Trend in Trend model is implemented in
<a href="../packages/TrendInTrend/index.html">TrendInTrend</a>.
        </li>
        <li>
          <i>
            Synthetic control
          </i>
          methods are implemented in
<a href="../packages/Synth/index.html">Synth</a>
          using a group method for comparative case studies,
<a href="../packages/microsynth/index.html">microsynth</a>
          for micro- and meso-level data, and
<a href="../packages/gsynth/index.html">gsynth</a>
          which extends synthetic control methods to multiple treated units and variable treatment periods.
        </li>
        <li>
          <i>
            Instrumental variable methods
          </i>
          are implemented in
<a href="../packages/ivmodel/index.html">ivmodel</a>,
<a href="../packages/ivpack/index.html">ivpack</a>
          (including power analysis, sensitivity analysis, and diagnostics),
<a href="../packages/bpbounds/index.html">bpbounds</a>
          (nonparametric bounds on ATE),
<a href="../packages/grf/index.html">grf</a>, and
<a href="../packages/estimatr/index.html">estimatr</a>.
<a href="../packages/ivmte/index.html">ivmte</a>
          provides a choice-theoretic interpretation to IV models using Marginal Treatment Effects to extrapolate from the compliers to estimate treatment effects for other subpopulations. Estimation of Marginal Treatment Effects using Local Instrumental Variables is implemented in
<a href="../packages/localIV/index.html">localIV</a>.
<a href="../packages/LARF/index.html">LARF</a>
          uses Local Average Response Functions for IV estimation of treatment effects with binary endogenous treatment and instrument.
<a href="../packages/icsw/index.html">icsw</a>
          implements inverse compliance score weighting for estimating average treatment effects with an instrumental variable.
        </li>
        <li>
          <i>
            Mediation analysis
          </i>
          can be performed with
<a href="../packages/cfma/index.html">cfma</a>
          (functional mediation analysis),
<a href="../packages/cit/index.html">cit</a>
          (likelihood-based tests),
<a href="../packages/MultisiteMediation/index.html">MultisiteMediation</a>
          for multisite trials,
<a href="../packages/DirectEffects/index.html">DirectEffects</a>
          to estimate the controlled direct effect when fixing a potential mediator to a specific value.
<a href="../packages/mediation/index.html">mediation</a>
          and
<a href="../packages/cfdecomp/index.html">cfdecomp</a>
          implement identification, inference and
<a href="../packages/mediation/index.html">mediation</a>
          additionally also provides sensitivity analysis for causal mediation effects. Linear mediation analysis for complex surveys using balanced repeated replication is implemented in
<a href="../packages/MedSurvey/index.html">MedSurvey</a>.
<a href="../packages/paths/index.html">paths</a>
          uses an imputation approach to estimate path-specific causal effects along with a set of bias formulas for conducting sensitivity analysis.
<a href="../packages/regmedint/index.html">regmedint</a>
          implements regression-based analysis with a treatment-mediator interaction term.
<a href="../packages/gma/index.html">gma</a>
          performs Granger mediation analysis for time series.
        </li>
        <li>
          <i>
            Under interference
          </i>, causal effect estimation can be achieved using
<a href="../packages/inferference/index.html">inferference</a>
          by inverse-probability weighted (IPW) estimators,
<a href="../packages/netchain/index.html">netchain</a>
          on collective outcomes by chain graph models approximating the projection of the full longitudinal data onto the observed data, and
<a href="../packages/interferenceCI/index.html">interferenceCI</a>
          provides some exact confidence intervals for treatment effects on a binary outcome in two-stage randomized experiments with interference
        </li>
        <li>
          Diagnostics and visualization for
          <i>
            Multiplicative Interaction Models
          </i>
          are implemented in
<a href="../packages/interflex/index.html">interflex</a>.
        </li>
        <li>
<a href="../packages/InvariantCausalPrediction/index.html">InvariantCausalPrediction</a>
          provides confidence intervals for causal effects, using data collected in different experimental or environmental conditions (with hidden variables), extensions to nonlinear models are implemented in
<a href="../packages/nonlinearICP/index.html">nonlinearICP</a>.
        </li>
        <li>
          <i>
            Regression discontinuity design
          </i>
          (RDD): density discontinuity testing (a.k.a. manipulation testing) is implemented in
<a href="../packages/rddensity/index.html">rddensity</a>, inference under local randomization is implemented in
<a href="../packages/rdlocrand/index.html">rdlocrand</a>, analysis with multiple cutoffs or scores is implemented in
<a href="../packages/rdmulti/index.html">rdmulti</a>, tools to perform power, sample size and minimum detectable effects (MDE) calculations are available in
<a href="../packages/rdpower/index.html">rdpower</a>, robust confidence interval construction and bandwidth selection are implemented in
<a href="../packages/rdrobust/index.html">rdrobust</a>,
<a href="../packages/RATest/index.html">RATest</a>
          provides a collection of randomization tests, including a permutation test for the continuity assumption of the baseline covariates in the sharp RDD.
        </li>
        <li>
          Regularized calibrated estimation of the average treatment effects (ATE) and local average treatment effects (LATE) is implemented in
<a href="../packages/RCAL/index.html">RCAL</a>.
        </li>
        <li>
<a href="../packages/WhatIf/index.html">WhatIf</a>
          offers easy-to-apply methods to evaluate counterfactuals that do not require sensitivity testing over specified classes of models.
        </li>
      </ul>
      <p>
        In addition,
<a href="../packages/causalsens/index.html">causalsens</a>,
<a href="../packages/hettx/index.html">hettx</a>,
<a href="../packages/dstat/index.html">dstat</a>
        provide functions for
        <i>
          sensitivity analyses
        </i>,
<a href="../packages/ui/index.html">ui</a>
        implements functions to derive uncertainty intervals and conduct sensitivity analysis for missing data and unobserved confounding.
<a href="../packages/BalanceCheck/index.html">BalanceCheck</a>
        provides tests to assess balancing of the treatment groups after matching,
<a href="../packages/cobalt/index.html">cobalt</a>
        generates balance tables and plots before and after covariate balancing, and
<a href="../packages/confoundr/index.html">confoundr</a>
        implements covariate-balance diagnostics for time-varying confounding.
      </p>
      <p>
        <a name="hte">
          <strong>
            Heterogeneous treatment effect estimation
          </strong>
        </a>
      </p>
      <p>
        Some of the above mentioned packages can also be used for heterogeneous treatment effect (HTE) estimation.
      </p>
      <ul>
        <li>
          <i>
            Bayesian approaches
          </i>
          for individual causal effect estimation and are available in several packages including
<a href="../packages/bartCause/index.html">bartCause</a>
          (based on Bayesian Additive Regression Trees) and
<a href="../packages/bcf/index.html">bcf</a>
          (Bayesian Causal Forst).
        </li>
        <li>
          <i>
            Efficacious treatment or population subset selection
          </i>
          exploiting treatment effect heterogeneity is implemented in
<a href="../packages/FindIt/index.html">FindIt</a>
          and
<a href="../packages/grf/index.html">grf</a>. This latter package supports missing covariate values using the Missing Incorporated in Attributes approach. Additionally the package
<a href="../packages/subdetect/index.html">subdetect</a>
          provides a test for the existence of a subgroup with enhanced treatment effect.
        </li>
        <li>
          Other approaches for personalized causal predictions are provided by
<a href="../packages/EffectTreat/index.html">EffectTreat</a>
          (exploiting correlation-based expressions), and for randomized data by
<a href="../packages/evalITR/index.html">evalITR</a>
          (it additionally allows for defining budget constraints) and by
<a href="../packages/SortedEffects/index.html">SortedEffects</a>
          (estimation and inference methods for sorted causal effects and classification analysis).
        </li>
        <li>
<a href="../packages/stepp/index.html">stepp</a>
          provides diagnostic plots to
          <i>
            explore treatment-covariate interactions
          </i>
          for survival or generalized linear models, applicable for continuous, binomial and count data arising from two or more treatment arms of a clinical trial.
        </li>
      </ul>
      <p>
        <a name="policy">
          <strong>
            Policy learning
          </strong>
        </a>
      </p>
      <ul>
        <li>
          Estimation of an optimal dynamic treatment regime using interactive Q-learning is implemented in
<a href="../packages/iqLearn/index.html">iqLearn</a>
          and
<a href="../packages/DynTxRegime/index.html">DynTxRegime</a>.
<a href="../packages/listdtr/index.html">listdtr</a>
          uses list-based rules to estimate the optimal dynamic treatment regime.
<a href="../packages/qLearn/index.html">qLearn</a>
          implements Q-learning for estimating optimal dynamic treatment regimes from two stage sequentially randomized trials.
<a href="../packages/quantoptr/index.html">quantoptr</a>
          implements estimation methods for optimal treatment regimes under marginal quantile, marginal mean, and mean absolute difference.
<a href="../packages/DTRreg/index.html">DTRreg</a>
          proposes different methods such as G-estimation, dynamic weighted OLS and Q-learning, as well as several variance estimation approaches, it can handle survival outcomes and continuous treatment variables.
<a href="../packages/QTOCen/index.html">QTOCen</a>
          provides methods for estimation of mean- and quantile-optimal treatment regimes from censored data.
        </li>
        <li>
<a href="../packages/smartsizer/index.html">smartsizer</a>
          provides a set of tools for determining the necessary sample size in order to identify the optimal dynamic treatment regime in a
          <i>
            sequential, multiple assignment, randomized trial (SMART)
          </i>,
<a href="../packages/DTRlearn2/index.html">DTRlearn2</a>
          implements estimators for general K-stage DTRs from SMARTs with Q-learning and several outcome-weighted learning methods.
        </li>
        <li>
          Efficient estimation of the population-level causal effects of stochastic interventions on a continuous-valued exposure is implemented in
<a href="../packages/txshift/index.html">txshift</a>
        </li>
        <li>
<a href="../packages/ITRLearn/index.html">ITRLearn</a>
          implements maximin-projection learning for recommending a meaningful and reliable individualized treatment regime, and also Q-learning and A-learning for estimating the group-wise contrast function.
<a href="../packages/ITRSelect/index.html">ITRSelect</a>
          implements sequential advantage selection and penalized A-learning for selecting important variables in optimal individualized (dynamic) treatment regime in both single-stage or multi-stage studies.
<a href="../packages/OTRselect/index.html">OTRselect</a>
          implements a penalized regression method that can simultaneously estimate the optimal treatment strategy and identify important variables for either censored or uncensored continuous response.
        </li>
        <li>
<a href="../packages/lpdensity/index.html">lpdensity</a>
          implements a density (and derivatives) estimator based on local polynomial regressions for counterfactual analysis, program evaluation, and policy recommendations.
        </li>
        <li>
          Estimation of treatment decision rules from a randomized clinical trial using Single-Index Models with Multiple-Links is implemented in
<a href="../packages/simml/index.html">simml</a>, Single-Index Models with a Surface-Link for optimizing individualized dose rules from an observational study is implemented in
<a href="../packages/simsl/index.html">simsl</a>.
        </li>
      </ul>
      <p>
        <a name="dag">
          <strong>
            Do-calculus and causal discovery
          </strong>
        </a>
      </p>
      <ul>
        <li>
          <i>
            Identifiability
          </i>
          is addressed by
<a href="../packages/causaleffect/index.html">causaleffect</a>
          and
<a href="../packages/dosearch/index.html">dosearch</a>
          providing algorithms to decide whether a causal effect is identifiable or not and by
<a href="../packages/CausalQueries/index.html">CausalQueries</a>
          that calculates arbitrary estimands for a given causal model;
<a href="../packages/causaloptim/index.html">causaloptim</a>
          provides tight bounds for a user defined DAG, query and constraints using a symbolic linear optimizer.
        </li>
        <li>
<a href="../packages/CIEE/index.html">CIEE</a>
          allows estimation of causal effects using estimating equations derived from a DAG.
        </li>
        <li>
          <i>
            Causal networks estimation
          </i>
          is implemented in
<a href="../packages/CompareCausalNetworks/index.html">CompareCausalNetworks</a>.
        </li>
        <li>
<a href="../packages/generalCorr/index.html">generalCorr</a>
          computes generalized correlations, partial correlations and
          <i>
            plausible causal paths
          </i>.
        </li>
        <li>
          Constrained maximum likelihood estimation of a DAG with intervention or observation data is implemented in
<a href="../packages/intdag/index.html">intdag</a>
          for linear Gaussian SEM.
        </li>
        <li>
<a href="../packages/pcalg/index.html">pcalg</a>
          provides a set of functions for causal structure learning: PC (for observational data without hidden variables), FCI and RFCI (for observational data with hidden variables), and GIES (for a mix of observational and interventional data) without hidden variables); and for causal inference using graphical models: the IDA algorithm, the Generalized Backdoor Criterion (GBC), the Generalized Adjustment Criterion (GAC); incorporating background knowledge is also possible.
        </li>
      </ul>
      <p>
        In addition,
<a href="../packages/dagitty/index.html">dagitty</a>
        provides a graphical tool to represent and analyze structural causal models (a web-based environment is also available:
        <a href="http://dagitty.net">
          DAGitty
        </a>).
      </p>
      <p>
        <a name="data">
          <strong>
            Specific types of data
          </strong>
        </a>
      </p>
      <ul>
        <li>
          <i>
            Longitudinal data / time series and censored data
          </i>: Causal effect estimation for time series is implemented in
<a href="../packages/CausalImpact/index.html">CausalImpact</a>
          (using a Bayesian approach),
<a href="../packages/confoundr/index.html">confoundr</a>
          (covariate-balance diagnostics), and
<a href="../packages/CausalMBSTS/index.html">CausalMBSTS</a>
          (for multivariate responses).
        </li>
        <li>
          Other packages, such as
<a href="../packages/PanelMatch/index.html">PanelMatch</a>
          implements a set of matching methods to time-series cross-sectional data, are dedicated to time series but also contain some (often basic) methods to handle missing data (see also
<a href="MissingData.html">MissingData</a>).
        </li>
        <li>
          <i>
            GWAS and SNPs
          </i>:
<a href="../packages/CKAT/index.html">CKAT</a>
          implements kernel based methods to jointly test genetic main effect and gene-treatment interaction effects for a set of SNPs.
<a href="../packages/epiGWAS/index.html">epiGWAS</a>
          provides functions to perform robust epistasis detection in GWAS.
        </li>
        <li>
          <i>
            Example data sets
          </i>
          to run frequent example problems from causal inference textbooks are accessible through the
<a href="../packages/causaldata/index.html">causaldata</a>
          package.
        </li>
        <li>
          Weighted, two-mode, and longitudinal networks analysis is implemented in
<a href="../packages/tnet/index.html">tnet</a>
        </li>
      </ul>
      <p>
        <a name="applications">
          <strong>
            Specific application fields
          </strong>
        </a>
      </p>
      <ul>
        <li>
          Behavior change sciences use specialized analyses and visualization tools implemented in
<a href="../packages/behaviorchange/index.html">behaviorchange</a>.
        </li>
        <li>
          Evaluation of biomarkers and estimation of treatment-biomarker effects can be done using tools from
<a href="../packages/bhm/index.html">bhm</a>
          (for biomarker-treatment effects).
        </li>
        <li>
          Qualitative Comparative Analysis type methods are implemented in
<a href="../packages/cna/index.html">cna</a>.
        </li>
        <li>
          <i>
            Mendelian randomization methods
          </i>
          used to examine causal effects related to certain genes are implemented in
<a href="../packages/MendelianRandomization/index.html">MendelianRandomization</a>,
<a href="../packages/mr.raps/index.html">mr.raps</a>
          for two-sample Mendelian randomization with summary statistics by using Robust Adjusted Profile Score,
<a href="../packages/MRPC/index.html">MRPC</a>
          for PC algorithm with the principle of Mendelian Randomization.
        </li>
        <li>
          Causal inference approaches in genetic systems exploit quantitative trait loci (QTL) genotypes to infer causal relationships among phenotypes: functions to simultaneously infer causal graphs and genetic architecture (acyclic and cyclic) are implemented in
<a href="../packages/qtlnet/index.html">qtlnet</a>.
        </li>
        <li>
<a href="../packages/tools4uplift/index.html">tools4uplift</a>
          uplift modeling aims at predicting the causal effect of an action such as a marketing campaign on a particular individual.
        </li>
        <li>
          <i>
            Coincidence analysis
          </i>
          through configurational comparative methods is provided by
<a href="../packages/cna/index.html">cna</a>.
        </li>
      </ul>
    </div>

  <h3>CRAN packages:</h3>
  <ul>
    <li><a href="../packages/allestimates/index.html">allestimates</a></li>
    <li><a href="../packages/autoCovariateSelection/index.html">autoCovariateSelection</a></li>
    <li><a href="../packages/bacondecomp/index.html">bacondecomp</a></li>
    <li><a href="../packages/BalanceCheck/index.html">BalanceCheck</a></li>
    <li><a href="../packages/bartCause/index.html">bartCause</a></li>
    <li><a href="../packages/BCEE/index.html">BCEE</a></li>
    <li><a href="../packages/bcf/index.html">bcf</a></li>
    <li><a href="../packages/BCHM/index.html">BCHM</a></li>
    <li><a href="../packages/beanz/index.html">beanz</a></li>
    <li><a href="../packages/behaviorchange/index.html">behaviorchange</a></li>
    <li><a href="../packages/bhm/index.html">bhm</a></li>
    <li><a href="../packages/blocksdesign/index.html">blocksdesign</a></li>
    <li><a href="../packages/Boptbd/index.html">Boptbd</a></li>
    <li><a href="../packages/borrowr/index.html">borrowr</a></li>
    <li><a href="../packages/bpbounds/index.html">bpbounds</a></li>
    <li><a href="../packages/BuyseTest/index.html">BuyseTest</a></li>
    <li><a href="../packages/causaldata/index.html">causaldata</a></li>
    <li><a href="../packages/causaldrf/index.html">causaldrf</a></li>
    <li><a href="../packages/causaleffect/index.html">causaleffect</a></li>
    <li><a href="../packages/CausalGAM/index.html">CausalGAM</a></li>
    <li><a href="../packages/CausalImpact/index.html">CausalImpact</a></li>
    <li><a href="../packages/CausalMBSTS/index.html">CausalMBSTS</a></li>
    <li><a href="../packages/causaloptim/index.html">causaloptim</a></li>
    <li><a href="../packages/CausalQueries/index.html">CausalQueries</a></li>
    <li><a href="../packages/causalsens/index.html">causalsens</a></li>
    <li><a href="../packages/causalweight/index.html">causalweight</a></li>
    <li><a href="../packages/CBPS/index.html">CBPS</a></li>
    <li><a href="../packages/cfdecomp/index.html">cfdecomp</a></li>
    <li><a href="../packages/cfma/index.html">cfma</a></li>
    <li><a href="../packages/CIEE/index.html">CIEE</a></li>
    <li><a href="../packages/cit/index.html">cit</a></li>
    <li><a href="../packages/cjoint/index.html">cjoint</a></li>
    <li><a href="../packages/CKAT/index.html">CKAT</a></li>
    <li><a href="../packages/clusteredinterference/index.html">clusteredinterference</a></li>
    <li><a href="../packages/CMatching/index.html">CMatching</a></li>
    <li><a href="../packages/cna/index.html">cna</a></li>
    <li><a href="../packages/cobalt/index.html">cobalt</a> (core)</li>
    <li><a href="../packages/CompareCausalNetworks/index.html">CompareCausalNetworks</a></li>
    <li><a href="../packages/confoundr/index.html">confoundr</a></li>
    <li><a href="../packages/Counterfactual/index.html">Counterfactual</a></li>
    <li><a href="../packages/dagitty/index.html">dagitty</a> (core)</li>
    <li><a href="../packages/did/index.html">did</a></li>
    <li><a href="../packages/DirectEffects/index.html">DirectEffects</a></li>
    <li><a href="../packages/dosearch/index.html">dosearch</a></li>
    <li><a href="../packages/DRDID/index.html">DRDID</a></li>
    <li><a href="../packages/drtmle/index.html">drtmle</a></li>
    <li><a href="../packages/dstat/index.html">dstat</a></li>
    <li><a href="../packages/DTRlearn2/index.html">DTRlearn2</a></li>
    <li><a href="../packages/DTRreg/index.html">DTRreg</a></li>
    <li><a href="../packages/DynTxRegime/index.html">DynTxRegime</a></li>
    <li><a href="../packages/eefAnalytics/index.html">eefAnalytics</a></li>
    <li><a href="../packages/EffectLiteR/index.html">EffectLiteR</a></li>
    <li><a href="../packages/EffectTreat/index.html">EffectTreat</a></li>
    <li><a href="../packages/endoSwitch/index.html">endoSwitch</a></li>
    <li><a href="../packages/epiGWAS/index.html">epiGWAS</a></li>
    <li><a href="../packages/estimatr/index.html">estimatr</a></li>
    <li><a href="../packages/evalITR/index.html">evalITR</a></li>
    <li><a href="../packages/experiment/index.html">experiment</a></li>
    <li><a href="../packages/FindIt/index.html">FindIt</a></li>
    <li><a href="../packages/FLAME/index.html">FLAME</a></li>
    <li><a href="../packages/generalCorr/index.html">generalCorr</a></li>
    <li><a href="../packages/gfoRmula/index.html">gfoRmula</a></li>
    <li><a href="../packages/gma/index.html">gma</a></li>
    <li><a href="../packages/GPSCDF/index.html">GPSCDF</a></li>
    <li><a href="../packages/grf/index.html">grf</a> (core)</li>
    <li><a href="../packages/gsynth/index.html">gsynth</a></li>
    <li><a href="../packages/hdm/index.html">hdm</a></li>
    <li><a href="../packages/hettx/index.html">hettx</a></li>
    <li><a href="../packages/hrIPW/index.html">hrIPW</a></li>
    <li><a href="../packages/icsw/index.html">icsw</a></li>
    <li><a href="../packages/idem/index.html">idem</a></li>
    <li><a href="../packages/inferference/index.html">inferference</a></li>
    <li><a href="../packages/intdag/index.html">intdag</a></li>
    <li><a href="../packages/interferenceCI/index.html">interferenceCI</a></li>
    <li><a href="../packages/interflex/index.html">interflex</a></li>
    <li><a href="../packages/InvariantCausalPrediction/index.html">InvariantCausalPrediction</a></li>
    <li><a href="../packages/ipcwswitch/index.html">ipcwswitch</a></li>
    <li><a href="../packages/ipw/index.html">ipw</a></li>
    <li><a href="../packages/ipwCoxCSV/index.html">ipwCoxCSV</a></li>
    <li><a href="../packages/ipwErrorY/index.html">ipwErrorY</a></li>
    <li><a href="../packages/iqLearn/index.html">iqLearn</a></li>
    <li><a href="../packages/ITRLearn/index.html">ITRLearn</a></li>
    <li><a href="../packages/ITRSelect/index.html">ITRSelect</a></li>
    <li><a href="../packages/ivmodel/index.html">ivmodel</a></li>
    <li><a href="../packages/ivmte/index.html">ivmte</a></li>
    <li><a href="../packages/ivpack/index.html">ivpack</a></li>
    <li><a href="../packages/LARF/index.html">LARF</a></li>
    <li><a href="../packages/listdtr/index.html">listdtr</a></li>
    <li><a href="../packages/localIV/index.html">localIV</a></li>
    <li><a href="../packages/lpdensity/index.html">lpdensity</a></li>
    <li><a href="../packages/ltmle/index.html">ltmle</a></li>
    <li><a href="../packages/Matching/index.html">Matching</a> (core)</li>
    <li><a href="../packages/MatchIt/index.html">MatchIt</a> (core)</li>
    <li><a href="../packages/MatchThem/index.html">MatchThem</a></li>
    <li><a href="../packages/mediation/index.html">mediation</a></li>
    <li><a href="../packages/MedSurvey/index.html">MedSurvey</a></li>
    <li><a href="../packages/MendelianRandomization/index.html">MendelianRandomization</a></li>
    <li><a href="../packages/microsynth/index.html">microsynth</a></li>
    <li><a href="../packages/mr.raps/index.html">mr.raps</a></li>
    <li><a href="../packages/MRPC/index.html">MRPC</a></li>
    <li><a href="../packages/MultisiteMediation/index.html">MultisiteMediation</a></li>
    <li><a href="../packages/netchain/index.html">netchain</a></li>
    <li><a href="../packages/nonlinearICP/index.html">nonlinearICP</a></li>
    <li><a href="../packages/optmatch/index.html">optmatch</a></li>
    <li><a href="../packages/OTRselect/index.html">OTRselect</a></li>
    <li><a href="../packages/PanelMatch/index.html">PanelMatch</a></li>
    <li><a href="../packages/paths/index.html">paths</a></li>
    <li><a href="../packages/pcalg/index.html">pcalg</a> (core)</li>
    <li><a href="../packages/preference/index.html">preference</a></li>
    <li><a href="../packages/pstest/index.html">pstest</a></li>
    <li><a href="../packages/PSweight/index.html">PSweight</a></li>
    <li><a href="../packages/qLearn/index.html">qLearn</a></li>
    <li><a href="../packages/qte/index.html">qte</a></li>
    <li><a href="../packages/qtlnet/index.html">qtlnet</a></li>
    <li><a href="../packages/QTOCen/index.html">QTOCen</a></li>
    <li><a href="../packages/quantoptr/index.html">quantoptr</a></li>
    <li><a href="../packages/RATest/index.html">RATest</a></li>
    <li><a href="../packages/RCAL/index.html">RCAL</a></li>
    <li><a href="../packages/rddensity/index.html">rddensity</a></li>
    <li><a href="../packages/rdlocrand/index.html">rdlocrand</a></li>
    <li><a href="../packages/rdmulti/index.html">rdmulti</a></li>
    <li><a href="../packages/rdpower/index.html">rdpower</a></li>
    <li><a href="../packages/rdrobust/index.html">rdrobust</a></li>
    <li><a href="../packages/regmedint/index.html">regmedint</a></li>
    <li><a href="../packages/rpsftm/index.html">rpsftm</a></li>
    <li><a href="../packages/sampleSelection/index.html">sampleSelection</a></li>
    <li><a href="../packages/seqDesign/index.html">seqDesign</a></li>
    <li><a href="../packages/simml/index.html">simml</a></li>
    <li><a href="../packages/simsl/index.html">simsl</a></li>
    <li><a href="../packages/smartsizer/index.html">smartsizer</a></li>
    <li><a href="../packages/SortedEffects/index.html">SortedEffects</a></li>
    <li><a href="../packages/SPORTSCausal/index.html">SPORTSCausal</a></li>
    <li><a href="../packages/stepp/index.html">stepp</a></li>
    <li><a href="../packages/subdetect/index.html">subdetect</a></li>
    <li><a href="../packages/Synth/index.html">Synth</a></li>
    <li><a href="../packages/ThreeArmedTrials/index.html">ThreeArmedTrials</a></li>
    <li><a href="../packages/tmle/index.html">tmle</a> (core)</li>
    <li><a href="../packages/tnet/index.html">tnet</a></li>
    <li><a href="../packages/tools4uplift/index.html">tools4uplift</a></li>
    <li><a href="../packages/TrendInTrend/index.html">TrendInTrend</a></li>
    <li><a href="../packages/twang/index.html">twang</a></li>
    <li><a href="../packages/twangContinuous/index.html">twangContinuous</a></li>
    <li><a href="../packages/txshift/index.html">txshift</a></li>
    <li><a href="../packages/ui/index.html">ui</a></li>
    <li><a href="../packages/WhatIf/index.html">WhatIf</a></li>
  </ul>

  <h3>Related links:</h3>
  <ul>
    <li>CRAN Task View: <a href="ClinicalTrials.html">ClinicalTrials</a></li>
    <li>CRAN Task View: <a href="Econometrics.html">Econometrics</a></li>
    <li>CRAN Task View: <a href="ExperimentalDesign.html">ExperimentalDesign</a></li>
    <li>CRAN Task View: <a href="MetaAnalysis.html">MetaAnalysis</a></li>
    <li>CRAN Task View: <a href="MissingData.html">MissingData</a></li>
    <li>CRAN Task View: <a href="Survival.html">Survival</a></li>
    <li><a href="http://dagitty.net">
  DAGitty: draw and analyze causal diagrams
</a></li>
  </ul>

</body>
</html>
